packages/dashboard-core-plugins/src/panels/IrisGridPanel.tsx

```diff
@@ -807,6 +818,7 @@ export class IrisGridPanel extends PureComponent<
       IrisGridModel.EVENT.TABLE_CHANGED,
       this.handleTableChanged
     );
+    model.addEventListener(IrisGridModel.EVENT.UPDATED, this.handleUpdated);
```

Comment:
I was incorrect about which event to listen to - this will only trigger when there's an update within the current viewport you're viewing, and isn't triggering when the table grows larger than the current viewport. We want to listen to the JS API `EVENT_SIZECHANGED`, which we'll need to wire up through `IrisGridModel`. 
Add a `SIZE_CHANGED` event in `IrisGridModel.EVENT`, then in `IrisGridTableModelTemplate` add an event listener on the table to listen for `EVENT_SIZECHANGED` and then emit the `SIZE_CHANGED` event, listen to that from here.